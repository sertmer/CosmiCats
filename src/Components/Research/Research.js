import React from 'react';
import './Research.scss'

export const Research = ({ caseStudy }) => {
  return (
    <article>
        <img 
          alt='Outer space from Nasas files'
          src={caseStudy.img}
        />
        <h2>nasa id</h2>
        <p>{caseStudy.id}</p>
        <h2>date taken</h2>
        <p>{caseStudy.dateCreated}</p>
        <h2>photographer</h2>
      </article>
  )
}

export default Research;