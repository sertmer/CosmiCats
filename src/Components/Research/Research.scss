@import '../../_scss/variables';
@import '../../_scss/mixins';

.research {
  display: flex;
  flex-direction: row;
  background: rgb(193,193,193);
  background: linear-gradient(
    0deg, 
    rgba(193,193,193,1) 0%, 
    rgba(245,245,245,1) 100%
  );
  padding: 1%;
  @include rounded-corners(5px);
  width: 70%;
}

.research-img {
  width: 60vw;
  height: 80vh;
  @include rounded-corners(5px);
}

.research-details {
  padding: 2%;
}