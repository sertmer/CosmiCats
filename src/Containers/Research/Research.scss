@import '../../_scss/variables';
@import '../../_scss/mixins';

.research {
  display: flex;
  flex-direction: row;
  background: rgb(193,193,193);
  background: linear-gradient(
    0deg, 
    rgba(193,193,193,1) 0%, 
    rgba(245,245,245,1) 100%
  );
  padding: 1%;
  @include rounded-corners(5px);
  width: 80%;
}

.research-img {
  width: 60vw;
  height: 86.5vh;
  @include rounded-corners(5px);
}

.research-details {
  padding: 2%;
}

.save-to-faves-btn {
  height: 10%;
  width: 90%;
  @include rounded-corners(5px);
  margin: 2%;
}