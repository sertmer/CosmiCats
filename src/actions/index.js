export const setUser = user => ({
  type: 'SET_USER',
  user
})

export const setNasaImages = images => ({
  type: 'SET_NASA_IMAGES',
  images
})

export const setCatImages = images => ({
  type: 'SET_CAT_IMAGES',
  images
})

export const addToEvidence = evidence => ({
  type: 'ADD_EVIDENCE',
  evidence
})