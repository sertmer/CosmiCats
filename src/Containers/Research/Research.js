import React from 'react';
import { connect } from 'react-redux';
import { addToEvidence } from '../../actions'
import './Research.scss';
import PropTypes from 'prop-types';

export const Research = ({ caseStudy, addToEvidence }) => {
  return (
    <article className='research'>
      <img className='research-img'
        alt='Outer space from Nasas files'
        src={caseStudy.img}
      />
      <div className='research-details'>
        <h2>nasa id</h2>
        <p>{caseStudy.id}</p>
        <h2>date taken</h2>
        <p>{caseStudy.dateCreated}</p>
        <button
            className='save-to-faves-btn'
            type='button' 
            onClick={() => addToEvidence(caseStudy)}
          > save to evidence
          </button>
      </div>
    </article>
  )
}

export const mapDispatchToProps = dispatch => ({
  addToEvidence: evidence => dispatch(addToEvidence(evidence))
})

export default connect(null, mapDispatchToProps)(Research);

Research.propTypes = {
  caseStudy: PropTypes.object,
  addToEvidence: PropTypes.func
}