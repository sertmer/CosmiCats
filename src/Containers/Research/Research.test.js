import React from 'react';
import { shallow } from 'enzyme';
import { addToEvidence } from '../../actions'
import { Research, mapDispatchToProps } from './Research';

describe('Research', () => {

  let wrapper
  let mockAddToEvidence
  let mockCaseStudy

  beforeEach(() => {
    mockCaseStudy = {
      img: 'url.com',
      id: 'ab1',
      dateCreated: '12-12-12'
    }
    wrapper = shallow(<Research caseStudy={mockCaseStudy}/>)
  })

  describe('Research unit tests', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the addToEvidence action', () => {
      const mockEvidence = {
        img: 'url.com',
        id: 'ab1',
        dateCreated: '12-12-12'
      }
  
      const mockDispatch = jest.fn()
      const mockAddToEvidenceAction = addToEvidence(mockEvidence)
      const mappedProps = mapDispatchToProps(mockDispatch)
  
      mappedProps.addToEvidence(mockEvidence)
  
      expect(mockDispatch).toHaveBeenCalledWith(mockAddToEvidenceAction)
    })
  })
})