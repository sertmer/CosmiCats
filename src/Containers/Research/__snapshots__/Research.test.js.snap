// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Research Research unit tests should match the snapshot 1`] = `
<h2>
  hey
</h2>
`;
